src/services/navigator/helpers.ts

```diff
@@ -301,3 +302,132 @@ export const buildRequest = (
 
   return [navigation, lastPathId || routeId, params];
 };
+
+/**
+ * Create a map of <id, element> from a list of nodes
+ */
+const nodesToMap = (
+  nodes: TypesLegacy.NodeList<TypesLegacy.Node>,
+): Map<string, TypesLegacy.Node> => {
+  const map: Map<string, TypesLegacy.Node> = new Map();
```

Comment:
does using `Map` has any advantage in this situation over a simple object?

Reply:
In other languages, the keys are a hash table which provides quick access by key. Maybe not necessary here. Let me know if I should change it over.

Reply:
Swapped for obj